cli/cmd/initnodes.go

```diff
@@ -398,3 +423,96 @@ func gethInit(ctx context.Context, cfg InitConfig, dir string) error {
+		// Copy the newly generated jwtsecret file from the renamed geth home path to the actual geth home path with the newly restored snapshot data.
+		jwtSecretSrcFilePath := origAbsClientStorageDir + "/jwtsecret"
+		jwtSecretDestFilePath := absClientStorageDir + "/jwtsecret"
+		if err := fileio.Copy(jwtSecretSrcFilePath, jwtSecretDestFilePath); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("❌ Error copying jwtsecret from %s to %s", origAbsClientStorageDir, absSnapshotOutputDir))
+		}
+
+		// Copy the newly generated nodekey file from the renamed geth home path to the actual geth home path with the newly restored snapshot data.
+		nodekeyFileSrcPath := origAbsClientStorageDir + "/nodekey"
+		nodekeyFileDestPath := absClientStorageDir + "/nodekey"
+		if err := fileio.Copy(nodekeyFileSrcPath, nodekeyFileDestPath); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("❌ Error copying nodekey from %s to %s", origAbsClientStorageDir, absSnapshotOutputDir))
+		}
+	case haloClientName:
+		chainStatusSrcPath := filepath.Join(homeDir, "chain_status.json")
+		chainStatusDestPath := filepath.Join(absClientStorageDir, "chain_status.json")
+		if err := fileio.Copy(chainStatusSrcPath, chainStatusDestPath); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("❌ Error copying chain_status.json from %s to %s", homeDir, absClientStorageDir))
+		}
+	}
+
+	return nil
+}
+
+func downloadLatestSnapshot(ctx context.Context, network string, clientName string, outputDir string) error {
+	snapshotArchive := getSnapshotBackupArchive(clientName)
+	bucketName := fmt.Sprintf("omni-%s-snapshots", network)
+	gcpCloudStorageURL := fmt.Sprintf("https://storage.googleapis.com/%s/%s", bucketName, snapshotArchive)
+	outputFilePath := filepath.Join(outputDir, snapshotArchive)
+
+	log.Info(ctx, fmt.Sprintf("🌐 Downloading latest snapshot for %s...", clientName))
+	if err := fileio.Download(ctx, gcpCloudStorageURL, outputFilePath); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("❌ Error downloading latest %s snapshot for %s", network, clientName))
+	}
+
+	if clientName == haloClientName {
+		log.Info(ctx, fmt.Sprintf("🌐 Downloading chain_status.json for %s...", clientName))
+		gcpCloudStorageChainStatusURL := fmt.Sprintf("https://storage.googleapis.com/%s/chain_status.json", bucketName)
+		chainStatusOutputFilePath := filepath.Join(outputDir, "chain_status.json")
+		if err := fileio.Download(ctx, gcpCloudStorageChainStatusURL, chainStatusOutputFilePath); err != nil {
+			return errors.Wrap(err, "❌ Error downloading latest chain_status.json for "+clientName)
+		}
+	}
+
+	return nil
+}
+
+func getSnapshotBackupArchive(clientName string) string {
+	return clientName + "_data.tar.lz4"
+}
+
+func getClientStorageDir(clientName string) (string, error) {
+	switch clientName {
+	case gethClientName:
+		return "geth/geth", nil
+	case haloClientName:
+		return "halo/data", nil
+	default:
+		return "", errors.New("unknown client name: %s. Possible options are: halo, geth", clientName)
```

Comment:
simplify, no need to provide possible options, since you are not handling user input, simply `errors.New("invalid client")`.

Reply:
Done.